### ISSUE TITLE
redundantSelf rule incorrectly removes self nested functions and index lookups inside autoclosure calls even when --selfrequired is present

### ISSUE THREAD

**Issue by stevelandeyasana** [nicklockwood/SwiftFormat#1066]
I have a Nimble test that uses code like this:

```swift
class MyTest: XCTestCase {
  var foo = 1
  func testThing() {
    let array = ["a", "b", "c"]
    expect(array[self.foo]).to(equal("b"))
  }
}
```

SwiftFormat incorrectly updates this code to remove the `self.`:

```swift
class MyTest: XCTestCase {
  var foo = 1
  func testThing() {
    let array = ["a", "b", "c"]
    expect(array[foo]).to(equal("b")) // oh no!
  }
}
```

There's a similar issue for nested function calls.

And now my code won't build. So I need to downgrade SwiftFormat, or disable the redundantSelf rule I had previously relied on. :-(

**Comment by stevelandeyasana** [comment]
Seems related to #993.

**Comment by nicklockwood** [comment]
Sorry about that - I'll get this fixed. In the meantime, you can disable the rule just for that line by using:

```swift
// swiftformat:disable:next redundantSelf
expect(array[self.foo]).to(equal("b"))
```

without disabling the rule altogether.

**Comment by stevelandeyasana** [comment]
Thanks for taking a look! I appreciate the suggestion but there are too many occurrences. I'll just wait for the fix and re-enable the rule. :-)

**Comment by nicklockwood** [comment]
FYI, I think this is a duplicate of https://github.com/nicklockwood/SwiftFormat/issues/1058 rather than #993

**Comment by nicklockwood** [comment]
@stevelandeyasana fixed in 0.48.18
